File: src/Framework/Output/RawFrameOutput.fs
namespace Animations.Output

open Animations
open System
open System.IO

type public RawFrameOutput(destinationFileFormat: string) =
    
    let mutable frameNumber: int = 0

    member this.Write(frame: Frame): unit = 
        File.WriteAllBytes(String.Format(destinationFileFormat, frameNumber), frame.Bytes)
        frameNumber <- frameNumber + 1

    interface IVideoOutput with
        member this.Dispose(): unit = ()
        member this.Write frame = this.Write frame
    

File: Animations.Framework/Output/RawFrameOutput.cs
using System.IO;
using Animations.Framework.Drawing;

namespace Animations.Framework.Output
{
    public class RawFrameOutput : IVideoOutput
    {
        public RawFrameOutput(string destinationFileFormat)
        {
            _destinationFileFormat = destinationFileFormat;
        }

        public void Dispose()
        {
        }

        public void Write(Frame frame)
        {
            File.WriteAllBytes(string.Format(_destinationFileFormat, _frameNumber++), frame.Bytes);
        }
        
        private int _frameNumber;
        private readonly string _destinationFileFormat;
    }
}